### ISSUE TITLE
`public_member_api_docs` ignores constructors in abstract interface classes

### ISSUE THREAD

**Issue by Pante** [dart-lang/linter#5034]
**Describe the issue**
`public_member_api_docs` does not flag constructors in abstract interface classes that are undocumented.

**To Reproduce**
```dart
abstract interface class Foo {
  factory Foo.a() = Bar;
  
  factory Foo.b() => Bar();

  Foo();
}

class Bar extends Foo {}
```

**Expected behavior**
The constructors should be flagged as undocumented, similar to how they are flagged in abstract & interface classes.

**Additional context**

**Comment by pq** [comment]
This is related to https://github.com/dart-lang/linter/issues/4526 but maybe we over-stepped in treating abstract interface classes as "effectively" private?

/cc @lrhn
